Better preflight

BryceStevenWilley opened the pull request:
Fixes #391. Also fixes #411.

A bigger change, because I tried to clean up how things split between the preflight yaml and the main yaml.  Open to splitting into smaller PRs if people want.

Biggest change: instead of stating what the PDF should look like in statements, we actually ask them in a series of yes no questions. Works pretty well, open to adding or removing questions as we see fit. 

![Screenshot from 2021-05-25 16-40-17](https://user-images.githubusercontent.com/6252212/119565501-04409a80-bd78-11eb-82d1-3af5a5f44eeb.png)

nonprofittechy commented:
Really nice change. I think use of radio buttons makes this much more helpful for authors.

Only thing:

1. If there are no custom names, then the `id: information about people` screen comes in a weird order. I would remove the information that isn't about custom names.
2. The information about custom names doesn't seem to get used in the "reserved names" debug table (the text isn't bolded).

![image](https://user-images.githubusercontent.com/7645641/119572048-2b9b6580-bd80-11eb-8098-96293470f26f.png)

Used this PDF and checked "rodent" and "Bobcat" as custom names.

[Test_custom_names.pdf](https://github.com/SuffolkLITLab/docassemble-assemblylinewizard/files/6542516/Test_custom_names.pdf)

I suggest saving that feature for a next release and just disabling the prompt about custom names for now, unless you see a quick fix for it. Otherwise it looks good to go.

nonprofittechy commented on docassemble/assemblylinewizard/data/questions/assembly_line.yml:
```diff
@@ -43,16 +43,18 @@ mandatory: True
   # Run pdf_field_tester.yml if form is a pdf.  
-  if template_upload[0].filename.endswith('pdf'):    
+  if template_upload[0].filename.endswith('pdf'):
+    if len(get_person_variables(all_fields, custom_only=True)) > 0:
```
This check doesn't seem to work correctly, at least not with this PDF
[NL_small_claims_repaired(1).pdf](https://github.com/SuffolkLITLab/docassemble-assemblylinewizard/files/6542558/NL_small_claims_repaired.1.pdf)

BryceStevenWilley replied:
That should be resolved with the new way of showing possible custom people.

BryceStevenWilley commented:
> The information about custom names doesn't seem to get used in the "reserved names" debug table (the text isn't bolded).

Yeah, I forgot to mention my thoughts on that earlier. The technical issue is that we don't look at or apply custom names until we already have combined multiple appearances of fields, and yes no check boxes, etc. However, at this point, we're specifically trying to make sure that each label in the interview is correct, and we don't want to just show them the combined fields, that would defeat the purpose of these preflight screens. 

Writing this, I'm realizing that we don't need to ask about and combine the custom people variable, we just need to mark the fields that are possibly custom, and include that in the table. I'll try that approach and see how it goes.

BryceStevenWilley commented:
> Writing this, I'm realizing that we don't need to ask about and combine the custom people variable, we just need to mark the fields that are possibly custom, and include that in the table. I'll try that approach and see how it goes.

Turned out pretty well, should have addressed all of your comments.

nonprofittechy commented:
Great! Looks like we just need to add the new pdf

BryceStevenWilley commented:
Thanks for catching that, always forget to add new files in the playground package.

nonprofittechy commented:
> Thanks for catching that, always forget to add new files in the playground package.

Yeah, I do the same thing. It might be a simple error we can add a test for, but maybe not.
